src/index.ts

```diff
@@ -0,0 +1,323 @@
+          //    are the ones that are still waiting to be fulfilled
+          const [pendingRequestedBeaconUpdateEvent] =
+            requestedBeaconUpdateEvents.filter(
+              (rbue) =>
+                !updatedBeaconEvents.some(
+                  (ub) => rbue.args!["requestId"] === ub.args!["requestId"]
+                )
+            );
+          if (!isNil(pendingRequestedBeaconUpdateEvent)) {
+            // 4. Check if RequestedBeaconUpdate event is awaiting fulfillment by
+            //    calling AirnodeRrp.requestIsAwaitingFulfillment with requestId
+            //    and check if beacon value is fresh enough and skip if it is
+
+            // `requestIdToTemplateId` is private so we must use AirnodeRrp instead
+            // const requestIsAwaitingFulfillment = await rrpBeaconServer.requestIdToTemplateId(
+            //   pendingRequest.args!["requestId"]
+            // );
+
+            const requestIsAwaitingFulfillment =
+              await airnodeRrp.requestIsAwaitingFulfillment(
+                pendingRequestedBeaconUpdateEvent.args!["requestId"]
+              );
+            //TODO: Add timestamp check?
+            if (requestIsAwaitingFulfillment) {
+              console.log(
+                "[INFO]\trequest is awaiting fulfillment. skipping update"
+              );
+              return;
+            }
+          }
+
+          // RrpBeaconServer expects that all parameters are defined in the template.
+          // There's a Jira issue to try adding user-defined parameters to the RrpBeaconServer.
+          // https://api3dao.atlassian.net/browse/A3P-48
+          // When using config.json.example we must pass a `from` parameter and the only
+          // way to get this request to work is by adding it as fixedParameter in the node config file
+          await rrpBeaconServer
+            .connect(keeperSponsorWallet)
+            .requestBeaconUpdate(
+              templateId,
+              requestSponsor,
+              requestSponsorWallet.address
+            );
+        }
+      });
+    })
+  );
+
+  await Promise.all(providerPromises);
+
+  const completedAt = new Date();
+  const durationMs = Math.abs(completedAt.getTime() - startedAt.getTime());
+  console.log(`[DEBUG]\tfinishing beaconUpdate after ${durationMs}ms...`);
+
+  const response = { ok: true, data: { message: "Beacon update requested" } };
+  return { statusCode: 200, body: JSON.stringify(response) };
+};
+
+export function loadAirkeeperConfig(): Config {
+  const configPath = path.resolve(`${__dirname}/../config/airkeeper.json`);
+  return JSON.parse(fs.readFileSync(configPath, "utf8"));
+}
+
+export function deriveKeeperSponsorWallet(
+  airnodeHdNode: ethers.utils.HDNode,
+  sponsorAddress: string,
+  provider: ethers.providers.Provider
+): ethers.Wallet {
+  const sponsorWalletHdNode = airnodeHdNode.derivePath(
+    `m/44'/60'/12345'/${deriveWalletPathFromSponsorAddress(sponsorAddress)}`
```

Comment:
Good catch!

But now I'm thinking, is it really a good idea to put a random number in the `change` part of the BIP44 derivation path? shouldn't that always be 0 (for ethereum at least)? It feels to me like we should come up with a different way to generate the `account` part for keeper sponsor wallets but I'm not completely sure tbh.

Reply:
Or maybe what you were suggesting is use the following path instead: m/44'/60'/0'/0/12345/x/y/z/t...

if so, would 12345 be the right number? should we use `(2**31) - 1` instead?

Reply:
My reasoning for having planned to use `change` as a sort of a "Airnode protocol ID" field is that it's redundant/meaningless for non-UTXO chains such as Ethereum in the first place, see https://www.reddit.com/r/Bitcoin/comments/6jamj6/why_does_bip44_define_a_change_path_rather_than/ so back when 
I already planned the current wallet derivation scheme around this so I'd rather follow through unless it's problematic 
```
export const deriveKeeperWalletPathFromSponsorAddress = (sponsorAddress: string): string => {
  const sponsorAddressBN = ethers.BigNumber.from(ethers.utils.getAddress(sponsorAddress));
  const paths = [];
  for (let i = 0; i < 6; i++) {
    const shiftedSponsorAddressBN = sponsorAddressBN.shr(31 * i);
    paths.push(shiftedSponsorAddressBN.mask(31).toString());
  }
  return `12345/${paths.join('/')}`; // use 12345 instead of 0
};
```
If we'll use 12345 or something else is entirely subjective and probably doesn't matter